### PR TITLE
apt: disable_components

### PR DESCRIPTION
**Pull request by dbungert**
Add API, and document for autoinstall, the ability to pass in components to
disable in apt.

### CODE REVIEW

File: subiquity/models/mirror.py [hunk 1]
```diff
@@ -51,12 +51,20 @@ def __init__(self):
         self.config = copy.deepcopy(DEFAULT)
         self.architecture = get_architecture()
         self.default_mirror = self.get_mirror()
+        self.disable_components = set()
 
-    def is_default(self):
+    def get_config(self):
+        # not the same as render() - this receives special processing rather
```
Comment (mwhudson) [inline subiquity/models/mirror.py:57]: This comment is out of date now given that render() returns {} now?

File: autoinstall-schema.json [hunk 2]
```diff
@@ -281,6 +281,9 @@
                 },
                 "sources": {
                     "type": "object"
+                },
+                "disable_components": {
+                    "type": "array"
```
Comment (mwhudson) [inline autoinstall-schema.json:286]: specifically it's an array of strings? Does it make sense to only accept restricted/main/universe/multiverse here?

**Review by mwhudson — APPROVED** [review]
Just a couple of comments.

File: autoinstall-schema.json (hunk 2 above)
Comment (dbungert) [reply autoinstall-schema.json]: good idea, fixed

File: subiquity/models/mirror.py (hunk 1 above)
Comment (dbungert) [reply subiquity/models/mirror.py]: done

**Comment by estan** [comment]
This may not be the place to ask, but I recently did an install of Ubuntu Server 20.04.4 LTS, and was surprised to see that multiverse was enabled by default in sources.list after the install. Is it supposed to be? I can't remember being given any choice about it during install, but I'm sure I did not enable it intentionally. Here's some log excerpts:

```
orexplore@oden:~$ sudo grep disable_components /var/log/installer/installer-journal.txt
Mar 08 16:08:12 ubuntu-server subiquity_log.3712[4158]: Handling apt to target /tmp/tmpdi7d_1np/mount with config {'disable_components': [], 'preserve_sources_list': False, 'primary': [{'arches': ['amd64', 'i386'], 'uri': 'http://se.archive.ubuntu.com/ubuntu'}, {'arches': ['default'], 'uri': 'http://ports.ubuntu.com/ubuntu-ports'}]}
orexplore@oden:~$ sudo grep multiverse /var/log/installer/installer-journal.txt
Mar 08 16:32:37 ubuntu-server subiquity_log.3712[4894]: Get:15 http://se.archive.ubuntu.com/ubuntu focal/multiverse amd64 Packages [144 kB]
Mar 08 16:32:37 ubuntu-server subiquity_log.3712[4894]: Get:16 http://se.archive.ubuntu.com/ubuntu focal/multiverse Translation-en [104 kB]
Mar 08 16:32:37 ubuntu-server subiquity_log.3712[4894]: Get:17 http://se.archive.ubuntu.com/ubuntu focal/multiverse amd64 c-n-f Metadata [9136 B]
Mar 08 16:32:37 ubuntu-server subiquity_log.3712[4894]: Get:27 http://se.archive.ubuntu.com/ubuntu focal-updates/multiverse amd64 Packages [23.8 kB]
Mar 08 16:32:37 ubuntu-server subiquity_log.3712[4894]: Get:28 http://se.archive.ubuntu.com/ubuntu focal-updates/multiverse Translation-en [7312 B]
Mar 08 16:32:37 ubuntu-server subiquity_log.3712[4894]: Get:29 http://se.archive.ubuntu.com/ubuntu focal-updates/multiverse amd64 c-n-f Metadata [580 B]
Mar 08 16:32:37 ubuntu-server subiquity_log.3712[4894]: Get:37 http://se.archive.ubuntu.com/ubuntu focal-backports/multiverse amd64 c-n-f Metadata [116 B]
Mar 08 16:32:38 ubuntu-server subiquity_log.3712[4894]: Get:47 http://se.archive.ubuntu.com/ubuntu focal-security/multiverse amd64 Packages [20.7 kB]
Mar 08 16:32:38 ubuntu-server subiquity_log.3712[4894]: Get:48 http://se.archive.ubuntu.com/ubuntu focal-security/multiverse Translation-en [5196 B]
Mar 08 16:32:38 ubuntu-server subiquity_log.3712[4894]: Get:49 http://se.archive.ubuntu.com/ubuntu focal-security/multiverse amd64 c-n-f Metadata [500 B]
Mar 08 16:35:27 ubuntu-server subiquity_log.3712[34036]: Get:8 http://se.archive.ubuntu.com/ubuntu focal/multiverse Translation-en [104 kB]
Mar 08 16:35:27 ubuntu-server subiquity_log.3712[34036]: Get:12 http://se.archive.ubuntu.com/ubuntu focal-updates/multiverse Translation-en [7312 B]
Mar 08 16:35:28 ubuntu-server subiquity_log.3712[34036]: Get:18 http://se.archive.ubuntu.com/ubuntu focal-security/multiverse Translation-en [5196 B]
orexplore@oden:~$
```

How can I get things into `disable_components` when doing a manual installation (to avoid having e.g. multiverse enabled)?

**Comment by dbungert** [comment]
Sadly, there is no UI point today that exposes disable_components.  You would have to use an autoinstall file to take advantage of that, and since you mention a manual install, I assume that autoinstall directives are not being used.

The relevant template for the sources.list should be here:
https://git.launchpad.net/livecd-rootfs/tree/live-build/functions#n866

**Comment by estan** [comment]
@dbungert Ah thanks for pointing me to that template, the actual cause for multiverse and restricted being enabled for me.

So there's no way to avoid these being enabled when running through the installer UI?

**Comment by dbungert** [comment]
Sorry, not at this time.  Might be interesting to add, though.

**Pull request merged**